### ISSUE TITLE
fileSystems."/".device with space messes up stab

### ISSUE THREAD

**Issue by gitfoxi** [NixOS/nixpkgs#4914]
I put a line like:

```
fileSystems."/".device = "/dev/disk/by-partlabel/Linux filesystem";
```

when I should have put:

```
 fileSystems."/".device = "/dev/disk/by-partlabel/Linux\ filesystem";
```

(Actually, that second one doesn't help either and I haven't figured out what does.)

When `nixos-install` writes `/etc/fstab` it doesn't get quoted (it probably should if that's possible) so you need to escape the space or at least error out. `nixos-rebuild switch` let me make this mistake too. Then it wouldn't let me fix my mistake and couldn't roll it back. Just kept saying it died in `/nix/var/nix/profiles/system/bin/switch-to-configuration` at line 239 which is where it validates the strings in your current `/etc/fstab` instead of validating the new one it's going to create. I had to mount my volume read/write to fix it in `/nix/store`. Is it a good idea? No.

So, to summarize, 3 things:
- Quote strings if possible
- Validate the new string
- Don't validate the old string we're trying to get rid of instead
## Update

You can supposedly escape spaces with `\040`. I tried that as well as `\\040` in case nix thinks I'm escaping the 0 and still no luck so I actually gave up and just renamed the stupid partitions.

**Comment by Profpatsch** [comment]
(triage) still a problem?

**Comment by Profpatsch** [comment]
Closing because no reply.
